admin/login.py

```diff
@@ -0,0 +1,135 @@
+"""
+    Copyright (c) 2020 - present NekrodNIK, Stepan Skriabin, rus-ai and other.
+    Look at the file AUTHORS.md(located at the root of the project) to get the full list.
+
+    This file is part of Morelia Server.
+
+    Morelia Server is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    Morelia Server is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public License
+    along with Morelia Server. If not, see <https://www.gnu.org/licenses/>.
+"""
+
+from pathlib import Path
+from loguru import logger
+
+from fastapi import APIRouter, Depends, Request
+from fastapi_login import LoginManager
+from fastapi.security import OAuth2PasswordRequestForm
+
+from configparser import ConfigParser
+
+from starlette.responses import HTMLResponse
+from mod import models
+from mod import lib
+import sqlobject as orm
+
+config = ConfigParser()
+config.read(Path(__file__).parent.parent / "config.ini")
```

Comment:
Это фишка Pathlib, хз как оно работает. Но в итоге мы получаем корректный путь к файлу(я это фишку увидел в сгенерированном setting.py django, мне понравилось, вот и использую). А по факту Pathlib - это os.path на максималках.

Reply:
Лады, пока апрув, но завтра проверю в винде, даже не думай

Reply:
Я сижу на винде, всё ок)))

Reply:
read(filenames, encoding=None)¶
Read and parse a filename or an iterable of filenames.

Files that cannot be opened are silently ignored; this is designed so that you can specify an iterable of potential configuration file locations (e.g. current directory, user’s home directory, systemwide directory), and all existing configuration files in the iterable will be read. A single filename may also be given.

Похоже ты зря директорию добавляешь, должно и просто с именем файла работать - оно ищет его везде.

Reply:
Не-а так не работает)))

Reply:
Тут вообще не понятно зачем использовать pathlib? Configparse и так находит файл.

Reply:
Не знаю... Просто зачем-то поставил эту конструкцию заодно.